### PR TITLE
Routing with redux actions

### LINKED ISSUE #47: Routing state is not synced correctly
Router is not correctly synced with the redux store, so you cannot see the routing state in the state, it alwayas appears as `null`.

### PR DESCRIPTION
**Pull request by alejandrodob**
Fixes #47. It also uses react router Link elements for navigation so the page is not reloaded when clicking on links. Navigation in the page is done via dispatching react-router-redux actions.

### CODE REVIEW

File: src/services/users/actions.js [hunk 1]
```diff
@@ -69,3 +69,7 @@ export function updateUser(data) {
 export function setPageNumber(pageNumber) {
   return setPageN(DOMAIN, pageNumber)
 }
+
+export function goToEdit(id) {
+  return push(`/my-account/users/edit/${id}`)
+}
```
Comment (acontreras89) [inline src/services/users/actions.js]: What's the difference between using `push(href)` and using a `<Link to={href} />`?

If there's none, I'd much rather use the 2nd form directly in the component.

**Comment by acontreras89** [comment]
If I'm not wrong, the only reason to use `Button`s instead of anchors (`<a>`) in user list items was how the CSS applies to each of them.

I don't know whether we can spare some minutes with someone in the design team to get this fixed, but I do think we should be moving to use anchors (and have them styled the same way as buttons) for buttons whose only purpose is to redirect.

**Comment by acontreras89** [comment]
Also, should we address #20 here as well?

File: src/services/users/actions.js (hunk 1 above)
Comment (alejandrodob) [inline src/services/users/actions.js]: `push` is to perform navigation actions programmatically so, yes, if we are going to dispatch the `goToLogin` action as a response to a user click (and we don't need to do anything else in response to that click) I absolutely agree with you that we should use Links.

**Pull request merged**